packages/smart-wallet/src/invitations.js

```diff
@@ -6,16 +6,31 @@ import { shape } from './typeGuards.js';
 // Ambient types. Needed only for dev but this does a runtime import.
 import '@agoric/zoe/exported.js';
 
+const { Fail } = assert;
+
+// A safety limit
+const MAX_PIPE_LENGTH = 2;
+
 /**
- * Supports three cases
- * 1. source is a contract (in which case this takes an Instance to look up in zoe)
- * 2. the invitation is already in your Zoe "invitation" purse so we need to query it
+ * Supports for cases:
+ * 1. source of invitation is a chain of calls starting with an agoricName
+ *     - each entry in the callPipe executes a call on the preceeding result
+ *     - if the methodName has a '.' then it is a special case expression, including:
+ *       - zoe.getPublicFacet - use the last result as an argument to E(zoe).getPublicFacet
+ *     - the end of the pipe is expected to return an Invitation
+ * 2. source is a contract (in which case this takes an Instance to look up in zoe)
+ * 3. the invitation is already in your Zoe "invitation" purse so we need to query it
  *    - use the find/query invitation by kvs thing
- * 3. continuing invitation in which the offer result from a previous invitation had an `invitationMakers` property
+ * 4. continuing invitation in which the offer result from a previous invitation had an `invitationMakers` property
  *
- * @typedef {ContractInvitationSpec | PurseInvitationSpec | ContinuingInvitationSpec} InvitationSpec
+ * @typedef {AgoricNameInvitationSpec | ContractInvitationSpec | PurseInvitationSpec | ContinuingInvitationSpec} InvitationSpec
```

Comment:
Do we need to look stuff up from agoric names now? I find this slightly confusing with the special case (thank you for documenting though). Can we just augment `ContractInvitationSpec` to use `callPipe` instead of just a single method name and args? We already can read the names from RPC.